src/nupic/utils/Random.cpp

```diff
@@ -58,7 +58,7 @@ Random::Random(UInt64 seed) {
   }
   // if seed is zero at this point, there is a logic error.
   NTA_CHECK(seed_ != 0);
-  gen.seed((unsigned int)seed_); //seed the generator
+  gen.seed(static_cast<unsigned int>(seed_)); //seed the generator
```

Comment:
store seed_ as uint? to avoid the cast at all

Reply:
> gen.seed(static_cast<unsigned int>(seed_)); //seed the generator
store seed_ as uint? to avoid the cast at all

When you worked on this you set this to UInt64.  Assumed you had a reason.
The argument to the constructor is UInt64.  Might break something if we change this.  Lets leave it for another PR.